azure-eventhubs-eph/src/main/java/com/microsoft/azure/eventprocessorhost/AzureStorageCheckpointLeaseManager.java

```diff
@@ -355,34 +423,48 @@ private AzureBlobLease getLeaseSync(String partitionId, BlobRequestOptions optio
     }
 
     @Override
-    public Iterable<Future<Lease>> getAllLeases() throws IllegalEntityException
+    public List<CompletableFuture<Lease>> getAllLeases(String[] partitionIds)
```

Comment:
>getAllLeases [](start = 42, length = 12)

use ListBlobs API
- which, then, the # of calls - can be optimized by storage

[https://docs.microsoft.com/en-us/java/api/com.microsoft.azure.storage.blob._cloud_blob_directory.listblobs](https://docs.microsoft.com/en-us/java/api/com.microsoft.azure.storage.blob._cloud_blob_directory.listblobs)